src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -65,30 +74,55 @@ public MuleConfigGenerator(List<ApikitMainFlowContainer> apis, List<GenerationMo
-        updateApikitConfig(api, muleConfig);
-        configs.add(muleConfig);
+        MuleConfig mainMuleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
+        Optional<MuleConfig> apikitConfigMuleConfig = updateApikitConfig(api, mainMuleConfig);
+        muleConfigs.add(mainMuleConfig);
+        addApikitConfig(muleConfigs, apikitConfigMuleConfig);
       });
     } else {
-      Set<MuleConfig> muleConfigs = new HashSet<>();
-      ApikitMainFlowContainer api = flowEntries.get(0).getApi();
-      MuleConfig muleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
-
+      ApikitMainFlowContainer api = flowEntries.stream().findFirst().get().getApi();
+      MuleConfig mainMuleConfig = api.getMuleConfig() != null ? api.getMuleConfig() : createMuleConfig(api);
+      Optional<MuleConfig> apikitConfig = updateApikitConfig(api, mainMuleConfig);
       for (GenerationModel flowEntry : flowEntries) {
         Element apikitFlowScope = new APIKitFlowScope(flowEntry, isMuleEE()).generate();
-
-        int newFlowPositionIndex = getLastFlowIndex(muleConfig.getContentAsDocument()) + 1;
-        muleConfig.getContentAsDocument().getRootElement().getContent().add(newFlowPositionIndex, apikitFlowScope);
-        muleConfig.addFlow(new Flow(apikitFlowScope));
-        updateApikitConfig(api, muleConfig);
-        muleConfigs.add(muleConfig);
+        int newFlowPositionIndex = getLastFlowIndex(mainMuleConfig.getContentAsDocument()) + 1;
+        mainMuleConfig.getContentAsDocument().getRootElement().getContent().add(newFlowPositionIndex, apikitFlowScope);
+        mainMuleConfig.addFlow(new Flow(apikitFlowScope));
+        muleConfigs.add(mainMuleConfig);
+        addApikitConfig(muleConfigs, apikitConfig);
       }
-      configs.addAll(muleConfigs);
     }
-    return configs;
+    return new ArrayList<>(muleConfigs);
+  }
+
+  private void addApikitConfig(Set<MuleConfig> muleConfigs, Optional<MuleConfig> apikitConfigMuleConfig) {
+    if (apikitConfigMuleConfig.isPresent()) {
+      muleConfigs.add(apikitConfigMuleConfig.get());
+    }
+  }
+
+  private MuleConfig retrieveApikitConfigMuleConfig(MuleConfig mainMuleConfig) {
+    MuleConfig apikitConfigMuleConfig = mainMuleConfig;
+    Optional<MuleConfig> apikitConfigMuleConfigOptional = Optional.empty();
+
+    if (muleConfigsInApp.size() > 1) {
+      String configRef = apikitConfigMuleConfig.getMainFlows().stream().findFirst().get().getApikitRouter().getConfigRef();
+      Stream<MuleConfig> apikitConfigurations = muleConfigsInApp.stream().filter(muleConfig -> muleConfig.getContentAsDocument()
+          .getRootElement().getChild("config", APIKitTools.API_KIT_NAMESPACE.getNamespace()) != null);
+      apikitConfigMuleConfigOptional =
+          apikitConfigurations.filter(configuration -> filterConfigurations(configRef, configuration)).findFirst();
+    }
+    return apikitConfigMuleConfigOptional.orElse(apikitConfigMuleConfig);
```

Comment:
return Optional.empty()

Reply:
return null instead is ok?

Reply:
After running tests figured out that `return mainMuleConfig` is required to work.